app/src/components/followButton/FollowButton.tsx

```diff
@@ -22,30 +24,55 @@ function FollowButton({
     selectIsAConnection(connectionId),
     shallowEqual
   );
+  const isAPendingConnectionRequest = useSelector(
+    selectIsAPendingConnectionRequest(connectionId),
+    shallowEqual
+  );
+
   const currentUserId = useSelector(selectCurrentUserId, shallowEqual);
   const isMe = connectionId === "me" || connectionId === currentUserId;
   const handleAddConnection = ({ isTeamMate }: { isTeamMate: boolean }) => {
-    dispatch(addConnectionAsync({ connectionId, isTeamMate }));
+    dispatch(requestAddConnectionAsync({ connectionId, isTeamMate }));
+  };
+
+  const handleCancelAddConnection = () => {
+    // Do something
   };
   return !isAConnection && !isMe ? (
     <div className={`Follow-button ${className || ""}`}>
-      <OptionsMenu
-        className="Follow-button__options-menu"
-        buttons={{
-          [`Add ${connectionName} as a connection`]: {
-            action: () => {
-              handleAddConnection({ isTeamMate: false });
+      {!isAPendingConnectionRequest && (
+        <OptionsMenu
+          className="Follow-button__options-menu"
+          buttons={{
+            [`Add ${connectionName} as a connection`]: {
+              action: () => {
+                handleAddConnection({ isTeamMate: false });
+              },
+            },
+            [`Add ${connectionName} as a team mate`]: {
+              action: () => {
+                handleAddConnection({ isTeamMate: true });
+              },
             },
-          },
-          [`Add ${connectionName} as a team mate`]: {
-            action: () => {
-              handleAddConnection({ isTeamMate: true });
+          }}
+        >
+          <span className="Follow-button__text">+ Follow</span>
+        </OptionsMenu>
+      )}
+      {isAPendingConnectionRequest && (
+        <OptionsMenu
+          className="Follow-button__options-menu"
+          buttons={{
+            [`Cancel pending connection request to ${connectionName}`]: {
+              action: () => {
+                handleCancelAddConnection();
+              },
             },
-          },
-        }}
-      >
-        <span className="Follow-button__text">+ Follow</span>
-      </OptionsMenu>
+          }}
+        >
```

Comment:
This works now, but I'm thinking a better approach would be to replace the "Follow" button for the user with the "cancel" option.